### DISCUSSION TITLE
Redeem promo codes from Play store (outside of app) in Android purchaseUpdatedListener is not called with receipt.

### DISCUSSION THREAD

**Discussion by nermin-sehic-hysp** [hyochan/react-native-iap#2549] [Q&A]
Has anyone encountered issues with redeeming promo codes from the Play Store outside of an Android app, and not receiving callbacks from the `purchaseUpdatedListener` in either of the app states? What is best practice to handle promo code that is redeemed outside of app? Help needed.

**Comment by muhammadsaadullah1** [comment]
share your code how you are redeeming please

**Comment by nishita-hyperspace** [comment]
<img width="980" alt="Screenshot 2023-09-29 at 10 41 49 AM" src="https://github.com/dooboolab-community/react-native-iap/assets/116154920/1f13ee3c-36c6-4572-bd29-1f89690bee32">
